crates/sui-framework/sources/governance/validator_set.move

```diff
@@ -522,9 +546,12 @@ module sui::validator_set {
         while (!vector::is_empty(&self.pending_removals)) {
             let index = vector::pop_back(&mut self.pending_removals);
             let validator = vector::remove(&mut self.active_validators, index);
-            table::remove(&mut self.staking_pool_mappings, staking_pool_id(&validator));
+            let validator_pool_id = staking_pool_id(&validator);
+            table::remove(&mut self.staking_pool_mappings, validator_pool_id);
             self.total_stake = self.total_stake - validator::total_stake_amount(&validator);
-            validator::destroy(validator, ctx);
+            // Destroy the validator object, but keep the staking pool around.
+            let staking_pool = validator::destroy(validator, ctx);
```

Comment:
Wondering if it makes sense to put the entire `Validator` in the table. It will cause more data to be stored on-chain for each inactive pool but it might be useful to still have access to the info (say name, image, project url) of an inactive validator for displaying? cc @Jordan-Mysten 

Also this way we may be able to move some extra logic dealing with inactive pools down to `validator`. What do you think?

Reply:
I was torn on whether we should keep the validator around, or simply keep the staking pool, so would be happy either way, and I'm more than happy to make this change if you think it would be useful to keep around :) 

I wound up going this direction because it (1) minimized the amount of the storage needed, and (2) the only `Validator` objects kicking around in the system are interesting, i.e., they are either active or could become active. Neither of these are super persuasive though...

Reply:
After thinking about it, I think it actually makes sense to keep the validators in this table as opposed to just the staking pool as long as there's money in its pool (for reporting/tracking purposes at the very least). However, the entry in this `inactive_validators` table is not guaranteed to be permanent -- once the validator's staking pool empties out it will be fully removed.

Reply:
@lxfind moving the conversation about `inactive_staking_pools` to this thread.

The reasoning in my mind is that this could make it easier to track the pool over time if the validator's metadata associated with the pool is kept around, e.g., you can track validator `YYY` started out with this initial stake, was deactivated at this epoch and has `XYZ` SUI remaining to be withdrawn from its staking pool. In general it seems safer/more futureproof to keep the `Validator` object here rather than strip it all the way down to its staking pool.

All this being said, I don't have super strong opinions here, so if there's a good reason one way or the other I'm more than happy to do that.

Reply:
Sounds good. No strong opinion :)